src/script/interpreter.cpp

```diff
@@ -2063,55 +2093,69 @@ bool VerifyScript(const CScript &scriptSig, const CScript &scriptPubKey,
             return false;
         }
     }
-    stackCopy = stack;
-    {
-        ScriptExecutionData execdata{scriptPubKey};
-        if (!EvalScript(stack, scriptPubKey, flags, checker, metrics, execdata,
-                        serror)) {
+    if (scriptPubKey.size() > 0 && scriptPubKey[0] == OP_SCRIPTTYPE) {
+        if (!VerifyScriptType(stack, scriptPubKey, flags, checker, metricsOut,
+                              serror)) {
             // serror is set
             return false;
         }
-    }
-    if (stack.empty()) {
-        return set_error(serror, ScriptError::EVAL_FALSE);
-    }
-    if (CastToBool(stack.back()) == false) {
-        return set_error(serror, ScriptError::EVAL_FALSE);
-    }
-
-    // Additional validation for spend-to-script-hash transactions:
-    if (scriptPubKey.IsPayToScriptHash()) {
-        // scriptSig must be literals-only or validation fails
-        if (!scriptSig.IsPushOnly()) {
-            return set_error(serror, ScriptError::SIG_PUSHONLY);
+        if (stack.empty()) {
```

Comment:
Not checking cleanstack here.

Reply:
Yes, it's checked here: https://github.com/cashweb/cashd/pull/129/files#diff-a0337ffd7259e8c7c9a7786d6dbd420c80abfa1afdb34ebae3261109d9ae3c19R2165